### PR TITLE
transmux: use stream based concatenation for clips

### PR DESCRIPTION
**Pull request by emranemran**
Occasionally, the mp4s of very short clips might stutter as it crosses segment boundaries. This was caused by incorrect PTS between segments which in turn is a result of us clipping and re-encoding the first and last segments "locally". Using a stream based concatenation instead of file based concatenation readjusts the timestamps and prevents these issues.

Note: Eventually, all mp4s will be generated using stream based concatenation if this works well for clipping.

### CODE REVIEW

File: video/transmux.go [hunk 1]
```diff
@@ -95,21 +97,89 @@ func MuxTStoFMP4(fmp4ManifestOutputFile string, inputs ...string) error {
...
-	// 1. create a .ts file for a given rendition
-	tsFile, err := os.Create(tsFileName)
-	if err != nil {
-		return totalBytes, fmt.Errorf("error creating file (%s) err: %w", tsFileName, err)
-	}
-	defer tsFile.Close()
-	// 2. for a given rendition, write all segment indices in ascending order to the single .ts file
-	for _, k := range segmentsList.GetSortedSegments() {
-		segBytes, err := tsFile.Write(segmentsList.SegmentDataTable[k])
+	if !useStreamBasedConcat {
```
Comment (thomshutt) [inline video/transmux.go:102]: This block is becoming pretty huge, could we split each part out into methods?

File: video/transmux.go [hunk 2]
```diff
@@ -95,21 +97,89 @@ func MuxTStoFMP4(fmp4ManifestOutputFile string, inputs ...string) error {
...
+			}
 		}
-		totalBytes = totalBytes + int64(segBytes)
+		// Create a .ts file for a given rendition
+		tsFile, err := os.Create(tsFileName)
+		if err != nil {
+			return totalBytes, fmt.Errorf("error creating file (%s) err: %w", tsFileName, err)
+		}
+		defer tsFile.Close()
+		defer os.Remove(tsFileName)
```
Comment (thomshutt) [inline video/transmux.go:164]: In the other disk-based pieces we did, we had logic on app startup to clear up any dangling files that got left behind from interrupted processes. Are we able to do that here?

File: video/transmux.go (hunk 1 above)
Comment (emranemran) [reply video/transmux.go:102]: Sure, I split out the ffmpeg bits into it's own separate fcn and left file management bits in the main fcn.

File: video/transmux.go (hunk 2 above)
Comment (emranemran) [reply video/transmux.go]: There's several deferred os.Removes in the top level fcn calls that ensures all dangling files are cleaned up if there's a panic/exit. But I didn't see any app startup logic to delete - can you point me to it?

**Comment by emranemran** [comment]
Updates: 
* addressed comments
* working on integration tests (might use a separate PR but trying to just get it done in this one)

**Comment by emranemran** [comment]
**Updates**:
- Added unit test since it made more sense than cucumber test for this case

File: video/transmux.go (hunk 2 above)
Comment (emranemran) [reply video/transmux.go]: Yeah, we're not doing this for clip/thumbnails/transmux/etc. Discussed with @mjh1 on a way to do this better and I think we'll create a top level dir that we can remove from init() all at one. Filed a separate ticket to get this done (VID-474).

**Review by mjh1 — APPROVED** [review]

**Pull request merged**